pyccel/symbolic/lambdify.py

```diff
@@ -32,7 +31,7 @@ def cse(expr):
     (vars_old, stmts) = map(list, zip(*ls))
     vars_new = []
     free_gl = expr.free_symbols
-    free_gl.update(expr.atoms(IndexedBase))
+    #free_gl.update(expr.atoms(IndexedBase)) #What should this be instead?
```

Comment:
Can you explain this comment?

Reply:
`expr.atoms(IndexedBase)` finds all the `IndexedBase` items in our code. We no longer have any of these items. I do not know enough about this function to know what should go here instead. I meant to ask @saidctb . Maybe it should find all objects with a `__getitem__` method?

Reply:
@saidctb Please have a look into this and create an issue if needed. Thanks!